### ISSUE TITLE
InDrops mode errors

### ISSUE THREAD

**Issue by srx3001** [sdparekh/zUMIs#42]
Hello there,
I have been using zUMIs successfully with facs data without any issues, but I have received some InDrops files, created by V3 protocol (so 4 reads ) and I am getting an error with it.

"[ W::sam_hdr_sanitise] Missing trailing newline on SAM header. Possibly truncated"

When I open the sam file (InDrops2.barcodelist.filtered.sort.sam) with head, it only has one header line and then empty space, but for some reason its 3GB in size.
Basically the SAM file just has this:

"@HD	VN:1.5	SO:queryname  "

My command line is as follows:
bash zUMIs-master.sh -r CDNA.fastq.gz -n InDrops2 -g star_reference_dir/**/ -a GTFs/Homo_sapiens.GRCh38.84.gtf -c 1-31 -m 1-6 -l 66 -p 64 -Y yes -F BC2.fastq.gz -L LIB.fastq.gz -f BC1.fastq.gz

Any hints why this might happen?
Best regards,
SK

**Comment by srx3001** [comment]
Update...
Looks like if I skip the filtering step and start from the mapping stage (-w mapping) I get all the outputs normally, the dgecounts matrix, etc.
There are no errors either.
Any idea what could cause this?

**Comment by cziegenhain** [comment]
Hey SK,

sorry that you didnt get a reply yet!
It could be a bug in the InDrops-specific fastq-filtering routine.
Would you mind uploading a small subset of reads from the input files (e.g 1 million) for us to reproduce the error?

**Comment by cziegenhain** [comment]
We have identified the problem: The InDrops-specific fastq-filtering routine failed to trim leading `@` in the read IDs when generating the barcode sam file.
This obviously lead to problems when samtools tried to sort the produced file.

We have now fixed the bug, please download the updated program from GitHub and try again.

Thanks for your patience!
